src/main.cpp

```diff
@@ -115,7 +113,7 @@ int main(int argc, const char** argv)
             ("no-ip-route", po::bool_switch(&no_ip_swap), "Disable encoding of routing via IP")
             ("link,l", po::value<unsigned int>(&link_failures), "Number of link-failures to model.")
             ("tos-reduction,r", po::value<size_t>(&tos), "0=none,1=simple,2=dual-stack,3=dual-stack+backup")
-            ("engine,e", po::value<size_t>(&engine), "0=no verification,1=moped,2=post*,3=pre*")
```

Comment:
@srba do we want to remove moped from the weighted analysis?
@MortenSchou / @dank100 does moped support weights?

Reply:
The old Moped version does not support weights. Moped version 2 has a weighted PDS library in C, but Moped v.2 does not parse PDS, so we would need to write our own interfacing code that uses the WPDS library. 

More info on Moped, including source code, is here: http://www2.informatik.uni-stuttgart.de/fmi/szs/tools/moped/

Reply:
On the topic of other implementations, there is also WPDS++ from University of Wisconsin. It's a C++ implementation of weighted pushdown system analysis with goals similar to Moped. Last modified in 2008. https://research.cs.wisc.edu/wpis/wpds/wpds++/

Reply:
I will put back moped, and modify it so it works for the new weighted PDA class, but just ignores the weights.

Reply:
Commit (6af8846) solves this.

Reply:
@petergjoel, do you want further changes regarding moped before merging?

Reply:
I am getting the following linking-errors:
```
/usr/bin/ld: CMakeFiles/aalwines.dir/main.cpp.o: in function `void pdaaal::Reducer::target_tos_prune<void, std::less<void> >(pdaaal::PDA<void, std::less<void> >&, unsigned long)':
main.cpp:(.text._ZN6pdaaal7Reducer16target_tos_pruneIvSt4lessIvEEEvRNS_3PDAIT_T0_EEm[_ZN6pdaaal7Reducer16target_tos_pruneIvSt4lessIvEEEvRNS_3PDAIT_T0_EEm]+0x518): undefined reference to `pdaaal::labels_t::noop_pre_filter(std::set<unsigned int, std::less<unsigned int>, std::allocator<unsigned int> > const&)'
/usr/bin/ld: CMakeFiles/aalwines.dir/main.cpp.o: in function `std::pair<unsigned long, unsigned long> pdaaal::Reducer::reduce<void, std::less<void> >(pdaaal::PDA<void, std::less<void> >&, int, unsigned long, unsigned long)':
main.cpp:(.text._ZN6pdaaal7Reducer6reduceIvSt4lessIvEEESt4pairImmERNS_3PDAIT_T0_EEimm[_ZN6pdaaal7Reducer6reduceIvSt4lessIvEEESt4pairImmERNS_3PDAIT_T0_EEimm]+0xc83): undefined reference to `pdaaal::labels_t::intersect(std::vector<unsigned int, std::allocator<unsigned int> > const&, unsigned long)'
/usr/bin/ld: CMakeFiles/aalwines.dir/main.cpp.o: in function `pdaaal::PDA<void, std::less<void> >::add_untyped_rule_impl(unsigned long, pdaaal::rule_t<void, std::less<void>, void>, bool, std::vector<unsigned int, std::allocator<unsigned int> > const&)':
main.cpp:(.text._ZN6pdaaal3PDAIvSt4lessIvEE21add_untyped_rule_implEmNS_6rule_tIvS2_vEEbRKSt6vectorIjSaIjEE[_ZN6pdaaal3PDAIvSt4lessIvEE21add_untyped_rule_implEmNS_6rule_tIvS2_vEEbRKSt6vectorIjSaIjEE]+0x148): undefined reference to `pdaaal::labels_t::merge(bool, std::vector<unsigned int, std::allocator<unsigned int> > const&, unsigned long)'
/usr/bin/ld: CMakeFiles/aalwines.dir/main.cpp.o: in function `pdaaal::PDAFactory<aalwines::Query::label_t, void, std::less<void> >::build_pda(pdaaal::PDAAdapter<aalwines::Query::label_t, void, std::less<void> >&, bool)':
main.cpp:(.text._ZN6pdaaal10PDAFactoryIN8aalwines5Query7label_tEvSt4lessIvEE9build_pdaERNS_10PDAAdapterIS3_vS5_EEb[_ZN6pdaaal10PDAFactoryIN8aalwines5Query7label_tEvSt4lessIvEE9build_pdaERNS_10PDAAdapterIS3_vS5_EEb]+0xd4d): undefined reference to `pdaaal::labels_t::merge(bool, std::vector<unsigned int, std::allocator<unsigned int> > const&, unsigned long)'
/usr/bin/ld: CMakeFiles/aalwines.dir/main.cpp.o: in function `pdaaal::PDAFactory<aalwines::Query::label_t, void, std::less<void> >::initialize_construction(pdaaal::PDAAdapter<aalwines::Query::label_t, void, std::less<void> >&, std::unordered_set<pdaaal::NFA<aalwines::Query::label_t>::state_t const*, std::hash<pdaaal::NFA<aalwines::Query::label_t>::state_t const*>, std::equal_to<pdaaal::NFA<aalwines::Query::label_t>::state_t const*>, std::allocator<pdaaal::NFA<aalwines::Query::label_t>::state_t const*> >&, std::vector<pdaaal::NFA<aalwines::Query::label_t>::state_t const*, std::allocator<pdaaal::NFA<aalwines::Query::label_t>::state_t const*> >&)':
main.cpp:(.text._ZN6pdaaal10PDAFactoryIN8aalwines5Query7label_tEvSt4lessIvEE23initialize_constructionERNS_10PDAAdapterIS3_vS5_EERSt13unordered_setIPKNS_3NFAIS3_E7state_tESt4hashISF_ESt8equal_toISF_ESaISF_EERSt6vectorISF_SK_E[_ZN6pdaaal10PDAFactoryIN8aalwines5Query7label_tEvSt4lessIvEE23initialize_constructionERNS_10PDAAdapterIS3_vS5_EERSt13unordered_setIPKNS_3NFAIS3_E7state_tESt4hashISF_ESt8equal_toISF_ESaISF_EERSt6vectorISF_SK_E]+0x4c5): undefined reference to `pdaaal::labels_t::merge(bool, std::vector<unsigned int, std::allocator<unsigned int> > const&, unsigned long)'
/usr/bin/ld: main.cpp:(.text._ZN6pdaaal10PDAFactoryIN8aalwines5Query7label_tEvSt4lessIvEE23initialize_constructionERNS_10PDAAdapterIS3_vS5_EERSt13unordered_setIPKNS_3NFAIS3_E7state_tESt4hashISF_ESt8equal_toISF_ESaISF_EERSt6vectorISF_SK_E[_ZN6pdaaal10PDAFactoryIN8aalwines5Query7label_tEvSt4lessIvEE23initialize_constructionERNS_10PDAAdapterIS3_vS5_EERSt13unordered_setIPKNS_3NFAIS3_E7state_tESt4hashISF_ESt8equal_toISF_ESaISF_EERSt6vectorISF_SK_E]+0x728): undefined reference to `pdaaal::labels_t::merge(bool, std::vector<unsigned int, std::allocator<unsigned int> > const&, unsigned long)'
/usr/bin/ld: main.cpp:(.text._ZN6pdaaal10PDAFactoryIN8aalwines5Query7label_tEvSt4lessIvEE23initialize_constructionERNS_10PDAAdapterIS3_vS5_EERSt13unordered_setIPKNS_3NFAIS3_E7state_tESt4hashISF_ESt8equal_toISF_ESaISF_EERSt6vectorISF_SK_E[_ZN6pdaaal10PDAFactoryIN8aalwines5Query7label_tEvSt4lessIvEE23initialize_constructionERNS_10PDAAdapterIS3_vS5_EERSt13unordered_setIPKNS_3NFAIS3_E7state_tESt4hashISF_ESt8equal_toISF_ESaISF_EERSt6vectorISF_SK_E]+0xd1d): undefined reference to `pdaaal::labels_t::merge(bool, std::vector<unsigned int, std::allocator<unsigned int> > const&, unsigned long)'
/usr/bin/ld: CMakeFiles/aalwines.dir/main.cpp.o:main.cpp:(.text._ZN6pdaaal10PDAFactoryIN8aalwines5Query7label_tEvSt4lessIvEE23initialize_constructionERNS_10PDAAdapterIS3_vS5_EERSt13unordered_setIPKNS_3NFAIS3_E7state_tESt4hashISF_ESt8equal_toISF_ESaISF_EERSt6vectorISF_SK_E[_ZN6pdaaal10PDAFactoryIN8aalwines5Query7label_tEvSt4lessIvEE23initialize_constructionERNS_10PDAAdapterIS3_vS5_EERSt13unordered_setIPKNS_3NFAIS3_E7state_tESt4hashISF_ESt8equal_toISF_ESaISF_EERSt6vectorISF_SK_E]+0x1365): more undefined references to `pdaaal::labels_t::merge(bool, std::vector<unsigned int, std::allocator<unsigned int> > const&, unsigned long)' follow
```

did you forget to add a library/dependency somewhere (or maybe a .cpp file)?